ValueCodable.playground/Sources/ValueCodable.swift

```diff
@@ -88,6 +88,22 @@ extension NSCoder {
         }
     }
 
+    public func decodeDynamicValue<Value, T0: ValueCodable, T1: ValueCodable>(ofType _: Value.Type, forKey key: String, _ t0: T0.Type, _ t1: T1.Type) -> Value? {
```

Comment:
Without variadic generics, would need to write an overload of this that took the number of concrete types you have